lib/data_logger/logger.ex

```diff
@@ -24,39 +24,27 @@ defmodule DataLogger.Logger do
   end
 
   @impl true
-  def init(%{prefix: prefix, options: options} = state) do
+  def init(%{prefix: prefix, options: options, module: destination} = state) do
     Registry.register(DataLogger.PubSub, prefix, nil)
 
-    options
-    |> Keyword.get(:send_async, false)
+    initialized_state = %{state | options: destination.initialize(options)}
+
+    initialized_state.options
```

Comment:
Last comment, I think we sometimes try to "overpipe" stuff.

You can rewrite this as:
```
if initialized_state.options[:send_async] do
  {:ok, Map.put_new(:tasks, %{})}
else
  {:ok, initialized_state}
end
```

Reply:
I know I can, but... I like pipes!